src/playback-watcher.js

```diff
@@ -347,10 +347,11 @@ export default class PlaybackWatcher {
     const buffered = this.tech_.buffered();
     const audioBuffered = sourceUpdater.audioBuffer ? sourceUpdater.audioBuffered() : null;
     const videoBuffered = sourceUpdater.videoBuffer ? sourceUpdater.videoBuffered() : null;
+    const media = this.media();
 
-    // verify that at least two segment durations have been
+    // verify that at least two segment durations or one part duration have been
     // appended before checking for a gap.
-    const twoSegmentDurations = (this.media().targetDuration - Ranges.TIME_FUDGE_FACTOR) * 2;
+    const durations = media.partTargetDuration ? media.partTargetDuration : media.targetDuration * 2;
```

Comment:
I lowered this to one part target duration. parts are so tiny that its almost impossible not be be off by a few ms.